### PR TITLE
Add flaky test attribute and mark `CanSeekForwardOnInterpolationFail` as flaky

### PR DESCRIPTION
**Pull request by frenzibyte**
Been failing for quite some time now.

### CODE REVIEW

File: osu.Framework.Tests/Clocks/InterpolatingFramedClockTest.cs [hunk 1]
```diff
@@ -156,7 +156,13 @@ public void NeverInterpolatesBackwardsOnInterpolationFail()
             Assert.Greater(interpolatedCount, 10);
         }
 
+        // always failing on single-thread macOS CI, failing when test asserts interpolating.IsInterpolating = false
```
Comment (peppy) [inline osu.Framework.Tests/Clocks/InterpolatingFramedClockTest.cs:159]: If it always fails, flaky probably won't work? It's only intended for intermittent failures.

Comment (frenzibyte) [reply osu.Framework.Tests/Clocks/InterpolatingFramedClockTest.cs:159]: Well sometimes it passes (see current CI status), so I wasn't sure about fully ignoring it. But I could easily do so, and make the ignore macOS-only using `Assert.Ignore`. Will leave it to your call.

Comment (peppy) [reply osu.Framework.Tests/Clocks/InterpolatingFramedClockTest.cs:159]: Dunno, I read your comment "always failing" and made the assumption that was the case. Change the comment to "Intermittently failing" i guess

Comment (peppy) [inline osu.Framework.Tests/Clocks/InterpolatingFramedClockTest.cs:159]:
```suggestion
        // Regularly failing on single-thread macOS CI, failing when test asserts interpolating.IsInterpolating = false
```

**Review by peppy — APPROVED** [review]

**Pull request merged**